diff --git a/packages/contract/node_modules/cashscript/dist/Transaction.js b/packages/contract/node_modules/cashscript/dist/Transaction.js
index f2a7ba5..36cf4a9 100644
--- a/packages/contract/node_modules/cashscript/dist/Transaction.js
+++ b/packages/contract/node_modules/cashscript/dist/Transaction.js
@@ -184,8 +184,7 @@ export class Transaction {
         if (manualTokenInputs.length > 0) {
             selectAllTokenUtxos(manualTokenInputs, this.outputs);
         }
-        const automaticTokenInputs = selectAllTokenUtxos(allUtxos, this.outputs);
-        const tokenInputs = manualTokenInputs.length > 0 ? manualTokenInputs : automaticTokenInputs;
+        const tokenInputs = manualTokenInputs.length > 0 ? manualTokenInputs : selectAllTokenUtxos(allUtxos, this.outputs);
         if (this.tokenChange) {
             const tokenChangeOutputs = createFungibleTokenChangeOutputs(tokenInputs, this.outputs, this.address);
             this.outputs.push(...tokenChangeOutputs);