diff --git a/node_modules/express-openapi-validator/dist/middlewares/openapi.request.validator.js b/node_modules/express-openapi-validator/dist/middlewares/openapi.request.validator.js
index 013a58b..61cdc8d 100644
--- a/node_modules/express-openapi-validator/dist/middlewares/openapi.request.validator.js
+++ b/node_modules/express-openapi-validator/dist/middlewares/openapi.request.validator.js
@@ -84,7 +84,7 @@ class RequestValidator {
             }
             const cookies = req.cookies
                 ? Object.assign(Object.assign({}, req.cookies), req.signedCookies) : undefined;
-            const valid = validator(Object.assign(Object.assign({}, req), { cookies }));
+            const valid = validator(Object.assign(Object.assign({}, req), { cookies, headers: {...req.headers} }));
             if (valid) {
                 next();
             }
